apps/glusterfs/block_volume_entry_create.go

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"github.com/boltdb/bolt"
+	"github.com/heketi/heketi/executors"
+	"github.com/lpabon/godbc"
+)
+
+func (v *BlockVolumeEntry) createBlockVolume(db *bolt.DB,
+	executor executors.Executor, blockHostingVolumeId string) error {
+
+	godbc.Require(db != nil)
+	godbc.Require(blockHostingVolumeId != "")
+
+	vr, host, err := v.createBlockVolumeRequest(db, executor,
+		blockHostingVolumeId)
+	if err != nil {
+		return err
+	}
+
+	blockVolumeInfo, err := executor.BlockVolumeCreate(host, vr)
+	if err != nil {
+		return err
+	}
+
+	v.Info.BlockVolume.Iqn = blockVolumeInfo.Iqn
+	v.Info.BlockVolume.Hosts = blockVolumeInfo.BlockHosts
+	v.Info.BlockVolume.Lun = 0
+	v.Info.BlockVolume.Username = blockVolumeInfo.Username
+	v.Info.BlockVolume.Password = blockVolumeInfo.Password
```

Comment:
I am not sure what does clear text mean. Yes, this will be saved in the db. These are parameters returned from the gluster-block for the volume create.